mitosheet/src/components/footer/DataSheetTabActions.tsx

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Mito
+
+import React, { useEffect } from 'react';
+import MitoAPI, { getRandomId } from '../../api';
+import { GraphDataJSON, GraphID, UIState } from '../../types';
+import Dropdown from '../elements/Dropdown';
+import DropdownItem from '../elements/DropdownItem';
+import { ModalEnum } from '../modals/modals';
+import { TaskpaneType } from '../taskpanes/taskpanes';
+
+/*
+    Helper function for finding all of the graph tab names
+    that are created from a given sheet index
+*/
+export const getGraphTabNamesAndIDsFromSheetIndex = (sheetIndex: number, graphDataJSON: GraphDataJSON): 
+({graphTabName: string, graphID: GraphID})[] => {
+    // Filter to only grapsh with the sheetIndex, and then get a list of the graph tab names
+    const filteredGraphDataJSON: GraphDataJSON = Object.fromEntries(Object.entries(graphDataJSON).filter(([, graphData]) => {
+        return graphData.graphParams.graphCreation.sheet_index === sheetIndex
+    }))
+
+    return Object.entries(filteredGraphDataJSON).map(([graphID, graphData]) => {
+        return {graphTabName: graphData.graphTabName, graphID: graphID}
+    })
+} 
+
+/*
+    Displays a set of actions one can perform on a sheet tab, including
+    deleting, duplicating, or renaming.
+*/
+export default function SheetTabActions(props: {
+    setDisplayActions: React.Dispatch<React.SetStateAction<boolean>>,
+    setIsRename: React.Dispatch<React.SetStateAction<boolean>>;
+    setUIState: React.Dispatch<React.SetStateAction<UIState>>;
+    closeOpenEditingPopups: () => void;
+    mitoAPI: MitoAPI
+    sheetIndex: number
+    graphDataJSON: GraphDataJSON
+}): JSX.Element {
+
+    // Log opening the sheet tab actions
+    useEffect(() => {
+        void props.mitoAPI.log(
+            'clicked_sheet_tab_actions',
+            {
+                tab_type: 'data',
+                sheet_index: props.sheetIndex
+            }
+        )
+    }, [])
+
+    const onDelete = async (): Promise<void> => {
+        const dependantGraphTabNamesAndIDs = getGraphTabNamesAndIDsFromSheetIndex(props.sheetIndex, props.graphDataJSON)
+        
+        if (dependantGraphTabNamesAndIDs.length > 0) {
+            props.setUIState(prevUIState => {
```

Comment:
Yeah, there more work that we need to do if we want to keep them here. We'd need to save the graphScript and graphHTML so it can still be accessed when they switch to that tab. We'd also need to make sure that the graph is not editable so you can't configure a graph that doesn't exists, etc. etc. 

Given I don't think there is really a benefit of this, but it requires more technical complexity, I vote we don't do it.